net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_api.inc

```diff
@@ -0,0 +1,230 @@
+<?php
+/*
+ * wg_api.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2021 Rubicon Communications, LLC (Netgate)
+ * Copyright (c) 2021 R. Christian McDonald
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// pfSense includes
+require_once('globals.inc');
+require_once('config.inc');
+require_once('pfsense-utils.inc');
+require_once('system.inc');
+require_once('pkg-utils.inc');
+require_once('service-utils.inc');
+require_once('util.inc');
+
+// WireGuard includes
+require_once('wg_globals.inc');
+
+// Get WireGuard status
+function wg_status() {
+	global $wgg;
+
+	$status = array();
+	exec("{$wgg['wg']} show", $status);
+
+	$output = implode("\n", $status);
+	return htmlspecialchars($output);
+
+}
+
+// Get WireGuard interface status
+function wg_interface_status() {
+	global $wgg;
+
+	$status = array();
+	exec("{$wgg['ifconfig']} -a -g {$wgg['if_group']}", $status);
+
+	$output = implode("\n", $status);
+	return htmlspecialchars($output);
+
+}
+
+// Get WireGuard kmod status
+function wg_kmod_status() {
+	global $wgg;
+
+	$output = exec("{$wgg['kldstat']} | grep {$wgg['kmod']}");
+	
+	return htmlspecialchars($output);
+
+}
+
+// Get wg and tools version
+function wg_version() {
+	global $wgg;
+
+	$packages = array();
+	exec("{$wgg['pkg']} info | grep -i {$wgg['pkg_name']}", $packages);

```

Comment:
Why is pkg binary path stored in `$wgg` array?  It's a regular system binary.  Also it's not needed to use grep, if you want to collect only installed version of a package you know the name you can use `pkg query %v pkg_name`

Reply:
> Always use full paths when calling an executable (e.g. /usr/bin/grep NOT grep)

Ref : https://docs.netgate.com/pfsense/en/latest/references/developer-style-guide.html#developer-rules

Reply:
Either way, I'm now using pkg query and was able to produce something a bit cleaner for everyone

![image](https://user-images.githubusercontent.com/3102039/115550040-62f58e80-a277-11eb-9942-d60e31db8b38.png)